pkg/kv/kvserver/client_mvcc_gc_test.go

```diff
@@ -68,3 +89,198 @@ func TestMVCCGCCorrectStats(t *testing.T) {
+
+	// Wait for the protection policy to apply to foo's range.
+	testutils.SucceedsSoon(t, func() error {
+		if len(repl.SpanConfig().GCPolicy.ProtectionPolicies) == 0 {
+			return errors.New("waiting for span config to apply")
+		}
+		require.NoError(t, repl.ReadProtectedTimestampsForTesting(ctx))
+		return nil
+	})
+
+	// Put some garbage in the stats, so it triggers the mvcc gc queue.
+	ms := repl.GetMVCCStats()
+	ms.KeyBytes = 16 * (1 << 20) // 16mb
+	ms.ValBytes = 32 * (1 << 20) // 32mb
+	ms.GCBytesAge = 48 * (1 << 20) * 100 * int64(time.Hour.Seconds())
+	repl.SetMVCCStatsForTesting(&ms)
+
+	// Run table foo's range through the MVCC GC queue until the GCThreshold is
+	// past the point keys and range deletion tombstone thereby classifying the
+	// range as empty.
+	db := tc.Server(0).DB()
+	testutils.SucceedsSoon(t, func() error {
+		require.NoError(t, store.ManualMVCCGC(repl))
+		var ba kv.Batch
+		ba.Header.MaxSpanRequestKeys = 1
+		ba.AddRawRequest(&kvpb.IsSpanEmptyRequest{
+			RequestHeader: kvpb.RequestHeader{
+				Key: tablePrefix, EndKey: tablePrefix.PrefixEnd(),
+			},
+		})
+		require.NoError(t, db.Run(ctx, &ba))
+		if !ba.RawResponse().Responses[0].GetIsSpanEmpty().IsEmpty() {
+			return errors.New("table span is not empty; re-run GC")
+		}
+		return nil
+	})
+
+	// Wait for the GC job to see that the span is empty and delete the zone
+	// config for the dropped table.
+	jobutils.WaitForJobToSucceed(t, runner, jobspb.JobID(jobID))
+
+	// Verify that the span config for the table is gone.
+	kvAccessor := s.SpanConfigKVAccessor().(spanconfig.KVAccessor)
+	kvSubscriber := s.SpanConfigKVSubscriber().(spanconfig.KVSubscriber)
+	sp := roachpb.Span{Key: tablePrefix, EndKey: tablePrefix.PrefixEnd()}
+	testutils.SucceedsSoon(t, func() error {
+		records, err := kvAccessor.GetSpanConfigRecords(ctx, spanconfig.Targets{
+			spanconfig.MakeTargetFromSpan(sp),
+		})
+		require.NoError(t, err)
+		if len(records) != 0 {
+			return errors.New("expected no span config records for table foo")
+		}
+		return nil
+	})
+
+	// Refresh the KVSubscriber to ensure it is caught up to now. We expect the
+	// cluster-wide protection to be combined with the fallback span config and
+	// applied to foo's empty range.
+	ptsReader := store.GetStoreConfig().ProtectedTimestampReader
+	require.NoError(t,
+		spanconfigptsreader.TestingRefreshPTSState(ctx, t, ptsReader, s.Clock().Now()))
+	ts, _, err := kvSubscriber.GetProtectionTimestamps(ctx, sp)
+	require.NoError(t, err)
+	require.Len(t, ts, 1)
+
+	ptsTime := hlc.Timestamp{WallTime: protectedTime}
+	testutils.SucceedsSoon(t, func() error {
+		require.NoError(t, store.ManualMVCCGC(repl))
+		require.NoError(t,
+			spanconfigptsreader.TestingRefreshPTSState(ctx, t, ptsReader, s.Clock().Now()))
+		threshold := repl.GetGCThreshold()
+		if !threshold.Equal(ptsTime.Prev()) {
+			if ptsTime.Prev().Less(threshold) {
+				t.Fatal("range GC threshold should never pass protected timestamp")
+			}
```

Comment:
You could use `require.False(t, ptsTime.Prev().Less(threshold), "range GC threshold should never pass protected timestamp")` here.